test/base/src/androidTest/kotlin/io/realm/test/shared/notifications/RealmListNotificationsTests.kt

```diff
@@ -110,21 +123,173 @@ class RealmListNotificationsTests : NotificationTests {
             }
 
             // Assertion after empty list is emitted
-            val firstEmittedList = channel.receive()
-            assertNotNull(firstEmittedList)
-            assertEquals(0, firstEmittedList.size)
+            channel.receive().let { listChange ->
+                assertIs<InitialList<*>>(listChange)
+
+                assertNotNull(listChange.list)
+                assertEquals(0, listChange.list.size)
+            }
+
+            // Assert a single range is reported
+            //
+            // objectListField = [<A, B>]
+            realm.writeBlocking {
+                val queriedContainer = findLatest(container)
+                val queriedList = queriedContainer!!.objectListField
+                queriedList.addAll(dataset)
+            }
+
+            channel.receive().let { listChange ->
+                assertIs<UpdatedList<*>>(listChange)
+
+                assertNotNull(listChange.list)
+                assertEquals(dataset.size, listChange.list.size)
+
+                assertIsChangeSet(
+                    (listChange as UpdatedList<*>),
+                    insertRanges = arrayOf(
+                        ListChange.Range(0, 2)
+                    )
+                )
+            }
+
+            // Assert multiple ranges are reported
+            //
+            // objectListField = [<C, D, E, F>, A, B, <G, H>]
+            realm.writeBlocking {
+                val queriedContainer = findLatest(container)
+                val queriedList = queriedContainer!!.objectListField
```

Comment:
Nitpick, I would move the `!!` to the line above, so the null-check trigger where the bug actually is created.